jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpanContext.java

```diff
@@ -77,8 +83,30 @@ public String getBaggageItem(String key) {
     return this.baggage;
   }
 
-  public long getTraceId() {
-    return traceId;
+  public String getTraceId() {
+    if (traceIdHigh == 0L) {
+      return Long.toHexString(traceIdLow);
+    }
+    final String hexStringHigh = Long.toHexString(traceIdHigh);
+    final String hexStringLow = Long.toHexString(traceIdLow);
+    if (hexStringLow.length() < 16) {
+      // left pad low trace id with '0'.
+      // In theory, only 12.5% of all possible long values will be padded.
+      // In practice, using Random.nextLong(), only 6% will need padding
+      final int padLeftLength = 16 - hexStringLow.length();
+      char[] zeroPadLeft = new char[padLeftLength];
+      Arrays.fill(zeroPadLeft, '0');
+      return hexStringHigh + new String(zeroPadLeft) + hexStringLow;
```

Comment:
This doesn't look 100% right to me: you are allocating a new array of zeroes every time you need to get the trace ID and the low value is lower than 16. Better would be something like:

```java
"0000000000000000".substring(hexStringLow.length()) + hexStringLow;
```

The zeroes are then treated as constant, which should have a better performance than dynamically allocating an array every time.

Reply:
An array is allocated in copyOfRange. But, I did not catch _copyOfRange was a java intrinsics :).

Reply:
Done